src/helpers.ts

```diff
@@ -0,0 +1,34 @@
+
+/**
+ * Finds the matching suffixes of the string, stripping off the non-matching starting characters.
+ * @param base
+ * @param comparee
+ */
+export function findIntersect(base: string, comparee: string): string {
+    const a = [...base].reverse();
+    const b = [...comparee].reverse();
+
+    // find the intersection and reverse it back into a string
+    const intersection: string[] = [];
+    let pos = 0;
+    // stop when strings at pos are no longer are equal
+    while (a[pos] === b[pos]) {
+        // if we reached the end or there isnt a value for that pos
+        // exit the while loop
+        if (!a[pos] || !b[pos]) { break; }
+        intersection.push(a[pos]);
+        pos++;
+    }
```

Comment:
this while loop is pretty dangerous and has caused a few bugs. originally I was considering recursion here with a exit condition of getting to the end of one array. it is fine the way it is though as I see you have quite a few more tests backing it thanks!